src/Microsoft.DotNet.Interactive.Tests/Server/SerializationTests.cs

```diff
@@ -147,6 +148,103 @@ public void All_event_types_are_tested_for_round_trip_serialization()
                 .BeEquivalentTo(eventTypes);
         }
 
+        [Fact]
+        public void Notebook_parse_request_can_be_deserialized()
+        {
+            var requestJson = GetTestFileContents();
+            var request = NotebookParseOrSerializeRequest.FromJson(requestJson);
+            using var _ = new AssertionScope();
+
+            request.Type.Should().Be("parse");
+            request.Id.Should().Be("the-id");
+            request.Extension.Should().Be(".dib");
+            request.DefaultLanguage.Should().Be("csharp");
+            request
+                .Should()
+                .BeOfType<NotebookParseRequest>()
+                .Which
+                .RawData
+                .Should()
+                .Equal(new byte[] { 0x01, 0x02, 0x03 });
+        }
+
+        [Fact]
+        public void Notebook_serialize_request_can_be_deserialized()
+        {
+            var requestJson = GetTestFileContents();
+            var request = NotebookParseOrSerializeRequest.FromJson(requestJson);
+            using var _ = new AssertionScope();
+
+            request.Type.Should().Be("serialize");
+            request.Id.Should().Be("the-id");
+            request.Extension.Should().Be(".dib");
+            request.DefaultLanguage.Should().Be("csharp");
+            request
+                .Should()
+                .BeOfType<NotebookSerializeRequest>();
+            var serializeRequest = (NotebookSerializeRequest)request;
+            serializeRequest.NewLine.Should().Be("\r\n");
+            serializeRequest.Document.Should().BeEquivalentTo(new InteractiveDocument(new[]
+            {
+                new InteractiveDocumentElement("csharp", "var x = 1;"),
+                new InteractiveDocumentElement("fsharp", "let y = 2"),
+            }));
+        }
+
+        [Fact]
+        public void Notebook_parse_response_can_be_serialized()
```

Comment:
Assent tests are best used for avoiding accidental breaking changes to the serialization format. I'd name the test accordingly because "can be serialized" doesn't indicate concern for the specific output.